lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1904,6 +1904,79 @@ def plot(self, xs, ys, *args, zdir='z', **kwargs):
 
     plot3D = plot
 
+    def fill_between(self, x1, y1, z1, x2, y2, z2, *, where=None, **kwargs):
+        """
+        Fill the area between two 3D curves.
+
+        The curves are defined by the points (*x1*, *y1*, *z1*) and
+        (*x2*, *y2*, *z2*). This creates one or multiple quadrangle
+        polygons that are filled. All points must be the same length N, or a
+        single value to be used for all points.
+
+        Parameters
+        ----------
+        x1, y1, z1 : float or 1D array-like
+            x, y, and z  coordinates of vertices for 1st line.
+
+        x2, y2, z2 : float or 1D array-like
+            x, y, and z coordinates of vertices for 2nd line.
+
+        where : array of bool (length N), optional
+            Define *where* to exclude some regions from being filled. The
+            filled regions are defined by the coordinates ``pts[where]``,
+            for all x, y, and z pts. More precisely, fill between ``pts[i]``
+            and ``pts[i+1]`` if ``where[i] and where[i+1]``. Note that this
+            definition implies that an isolated *True* value between two
+            *False* values in *where* will not result in filling. Both sides of
+            the *True* position remain unfilled due to the adjacent *False*
+            values.
+
+        **kwargs
+            All other keyword arguments are passed on to `.Poly3DCollection`.
+
+        Returns
+        -------
+        `.Poly3DCollection`
+            A `.Poly3DCollection` containing the plotted polygons.
+
+        """
+        had_data = self.has_data()
+        x1, y1, z1, x2, y2, z2 = cbook._broadcast_with_masks(x1, y1, z1, x2, y2, z2)
+
+        if where is None:
+            where = True
+        else:
+            where = np.asarray(where, dtype=bool)
+            if where.size != x1.size:
+                raise ValueError(f"where size ({where.size}) does not match "
+                                 f"size ({x1.size})")
+        where = where & ~np.isnan(x1)  # NaNs were broadcast in _broadcast_with_masks
+
+        polys = []
+        for idx0, idx1 in cbook.contiguous_regions(where):
+            x1slice = x1[idx0:idx1]
+            y1slice = y1[idx0:idx1]
+            z1slice = z1[idx0:idx1]
+            x2slice = x2[idx0:idx1]
+            y2slice = y2[idx0:idx1]
+            z2slice = z2[idx0:idx1]
+
+            if not len(x1slice):
+                continue
+
+            for i in range(len(x1slice) - 1):
```

Comment:
I think this is in relation to your "mode 2" comment, but I would have naively expected each contiguous region to be a single polygon. Adding additional black edges in the new test case doesn't seem ideal and I would have expected all black "edge colors" to only be around the rim of the fill_between.

I would have expected something like: iterate over all "1" points adding those to the (N*2, 3) list of points for this specific region/poly, move down to "2" and iterate over those in reverse.

I think it will also be much more performant to only have a few polygons instead of one for each quad number of polygons.

Reply:
Ok, I can add a `single_poly` boolean flag (better name welcome) to add back that "mode 2" behavior, as I agree that the single polygon covers a good amount of use cases, looks cleaner with our renderer, and should be more performant. But it's more limiting in the shapes than can be created in the general case, so I would prefer to have it default to `False`.

Reply:
No difference to the output there reversing the second list of points.